+bids/File2.m

```diff
@@ -0,0 +1,239 @@
+classdef File2
+
+  properties
+    prefix = ''     % bids prefix
+    extension = ''  % file extension
+    suffix = ''     % file suffix
+    entities = []   % list of entities
+    modality = ''   % name of file modality
+
+    bids_path = ''  % path within dataset
+    filename = ''   % bidsified name
+    json_filename = ''  % bidsified name for json file
+
+    entity_required = {}  % Required entities
+    entity_order = {}   % Expected order of entities
+    schema = []     % Schema used for given modality
+
+  end
+
+  properties (SetAccess = private)
+    changed = false
+  end
+
+  methods
+
+    function obj = File2(varargin)
+      args = inputParser;
+      charOrStruct = @(x) isstruct(x) || ischar(x);
+
+      args.addRequired('input_file', charOrStruct);
+      args.addParameter('use_schema', false, @islogical);
+      args.addParameter('tolerant', true, @islogical);
```

Comment:
Realising that overall bids-matlab is not super consistent in when something should be called as an argument or as "parameter: value" pair.

I think I would prefer to standardize this a bit before the first release. Will open an issue to plan this.

```matlab
BIDS = bids.layout(root, ...
                               use_schema,  ...
                               index_derivatives,  ...
                               tolerant,  ...
                               verbose)

query = bids.query(BIDS, filter)

bids.copy_to_derivative(BIDS, pipeline_name,...
                            out_path, ...
                            filters, ...
                            'unzip', true, ...
                            'force', false, ...
                            'skip_dep', false, ...
                            'use_schema', use_schema, ...
                            'verbose', true);

file = bids.File2(filename, 'use_schema', false ...
                                         'tolerant', true);

report = bids.report(BIDS, ...
                     'output_path', output_path, ...
                     'read_nifti', true, ...
                     'verbose', true);
```